iothub/service/src/Utilities/Fx.cs

```diff
@@ -57,7 +57,7 @@ internal static bool IsFatal(Exception ex)
 
         internal static bool ContainsAuthenticationException(Exception ex)
         {
-            while (ex != null)
+            if (ex != null)
```

Comment:
This can be simplified now:

```csharp
            return ex != null
                && (ex is AuthenticationException
                    || ContainsAuthenticationException(ex.InnerException));
```